### PR TITLE
build: Add `reset` script

### PR DESCRIPTION
**Pull request by tomi**
## Summary

We often face situations where a build will fail because some cache somewhere is messing up with the build. This PR adds a `nuke` script that can be used to delete all files that might be out of date (i.e. node_modules, dist files, .turbo cache, tsbuildinfo files, etc). It is a more destructive version from `clean`, which doesn't touch e.g. node_modules.

## Review / Merge checklist

- [ ] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) 
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included. 
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

### CODE REVIEW

**Review by netroy — COMMENTED** [review]
instead of maintaining a list of purgable files/folder in every package, could we use `git clean -xd` instead (with `--exclude `to skip deleting certain files/folders)?

**Comment by tomi** [comment]
> instead of maintaining a list of purgable files/folder in every package, could we use `git clean -xd` instead (with `--exclude `to skip deleting certain files/folders)?

@netroy I updated the script to be a bit more sophisticated. Added`.vscode/settings.json` and `.env` files to exclude. Anything else?

**Review by netroy — CHANGES_REQUESTED** [review]

File: scripts/reset.mjs [hunk 1]
```diff
@@ -0,0 +1,29 @@
+// Resets the repository by deleting all untracked files except for few exceptions.
```
Comment (netroy) [inline scripts/reset.mjs:1]: maybe we should add `process.env.FORCE_COLOR='1'` to pipe [subprocess TTY colors correctly](https://google.github.io/zx/known-issues#colors-in-subprocess)

**Review by netroy — COMMENTED** [review]
this should work, but has one minor issue that if someone runs `pnpm reset` before running `pnpm i`, this script will error out.

**Comment by tomi** [comment]
> this should work, but has one minor issue that if someone runs `pnpm reset` before running `pnpm i`, this script will error out.

Fixed this by ensuring zx is available. Not necessary ideal to run `pnpm install` to just delete the node_modules and run `pnpm install` again, but it works

**Review by netroy — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.58.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.58.0)

**Pull request merged**